pkg/kapp/cmd/tools/inspect_tree_view.go

```diff
@@ -178,14 +178,18 @@ func (a *assocSortingValue) uidOwnersStr() string {
 			foundRes, found := a.rsByUID[string(ref.UID)]
 			if found {
 				// only nest into first object that we find
-				uids = append([]string{string(ref.UID)}, uids...)
+				identifiers = append([]string{a.resIdentifier(foundRes)}, identifiers...)
 				nextRes = &foundRes
 				break
 			}
 		}
 	}
 
-	return "ref-" + strings.Join(uids, "/")
+	return "ref-" + strings.Join(identifiers, "/")
+}
+
+func (a *assocSortingValue) resIdentifier(resource ctlres.Resource) string {
+	return fmt.Sprintf("%s-%s", resource.Kind(), resource.Name())
```

Comment:
Replacing UIDs with a `<Kind>-<Name>` value, this results in deterministic sorting. In the examples I considered, the resources with same name and kind in different namespaces had a different `kapp.k14s.io/association` label as well and the got grouped separately. We could certainly consider namespaces too.

```suggestion
return fmt.Sprintf("%s-%s-%s", resource.Namespace(), resource.Name(), resource.Kind())
```